layer_factory/CMakeLists.txt

```diff
@@ -58,38 +54,12 @@ else()
     set (CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -Wpointer-arith -Wno-unused-function -Wno-sign-compare")
 endif()
 
-# Code-generate the layer_factory source files, and add a dependency target for them
-set(VLF_BASE_DEPENDENCIES
-    ${LVL_SCRIPTS_DIR}/vk.xml
-    ${LVL_SCRIPTS_DIR}/generator.py
-    ${LVL_SCRIPTS_DIR}/reg.py
-    ${VT_SCRIPTS_DIR}/layer_factory_generator.py
-    ${VT_SCRIPTS_DIR}/lvl_genvk.py
-    )
-add_custom_command(OUTPUT ${CMAKE_CURRENT_BINARY_DIR}/layer_factory.h
-    COMMAND ${PYTHON_CMD} ${VT_SCRIPTS_DIR}/lvl_genvk.py -registry ${LVL_SCRIPTS_DIR}/vk.xml layer_factory.h
-    DEPENDS ${VLF_BASE_DEPENDENCIES}
-    )
-add_custom_command(OUTPUT ${CMAKE_CURRENT_BINARY_DIR}/layer_factory.cpp
-    COMMAND ${PYTHON_CMD} ${VT_SCRIPTS_DIR}/lvl_genvk.py -registry ${LVL_SCRIPTS_DIR}/vk.xml layer_factory.cpp
-    DEPENDS ${VLF_BASE_DEPENDENCIES}
-    )
+run_vk_xml_generate(layer_factory_generator.py layer_factory.h)
+run_vk_xml_generate(layer_factory_generator.py layer_factory.cpp)
 add_custom_target(generate_vlf DEPENDS ${CMAKE_CURRENT_BINARY_DIR}/layer_factory.cpp ${CMAKE_CURRENT_BINARY_DIR}/layer_factory.h)
 set_target_properties(generate_vlf PROPERTIES FOLDER ${VULKANTOOLS_TARGET_FOLDER})
 
-# Layer Utils Library
-# For Windows, we use a static lib because the Windows loader has a fairly restrictive loader search
-# path that can't be easily modified to point it to the same directory that contains the layers.
-set(VKLAYER_UTILS_VLF_SOURCES ${V_LVL_ROOT_DIR}/layers/vk_layer_config.cpp ${V_LVL_ROOT_DIR}/layers/vk_layer_extension_utils.cpp ${V_LVL_ROOT_DIR}/layers/vk_layer_utils.cpp ${V_LVL_ROOT_DIR}/layers/vk_format_utils.cpp)
-if (WIN32)
-    add_library(VkLayer_utils_vlf STATIC ${VKLAYER_UTILS_VLF_SOURCES})
-else()
-    add_library(VkLayer_utils_vlf SHARED ${VKLAYER_UTILS_VLF_SOURCES})
-    if(INSTALL_LVL_FILES)
-        install(TARGETS VkLayer_utils_vlf DESTINATION $CMAKE_INSTALL_LIBDIR})
-    endif()
-endif()
-add_dependencies(VkLayer_utils_vlf generate_helper_files)
+add_dependencies(VkLayer_utils generate_helper_files)
```

Comment:
Do you need to add this dependency?

VkLayer_utils is a target that is built within LVL and so its dependencies should (already) be specified there.

Below, your layer targets defined in this file specify VkLayer_utils as their own dependencies, which is desired.

Reply:
Probably not, I'll pull it and retest.